product-service/src/handlers/getProductsById.ts

```diff
@@ -0,0 +1,26 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import 'source-map-support/register';
+import products from './productsList.json';
+
+export const getProductsById: APIGatewayProxyHandler = async (event) => {
+  console.log('getProductsById lambda invocation with event:', event);
+  console.log('Path Parameters:', event.pathParameters)
+
+  const { productId } = event.pathParameters;
+
+  const product = products.find(item => item.id === productId)
+
+  return !!product
+    ? {
+      statusCode: 200,
+      headers: {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Origin': '*',
+        'Access-Control-Allow-Credentials': true,
+      },
+      body: JSON.stringify(product)
+    }
+    : {
+      statusCode: 404
+    };
+};

```

Comment:
If you checked for absence of a product in an `if` statement and returned 404 there it would be more readable than a ternary operator.